lua/neotest-kotlin/src/output-parser.lua

```diff
@@ -93,16 +104,55 @@ M.parse_line = function(line, path, package)
 	}
 end
 
+---Determines all fully qualified classes in the provided file
+---@param file string
+---@return table<string, string>
+local function determine_all_classes_file(file)
+	if neotest.files.is_dir(file) then
+		error(string.format("determine_all_classes_file only operates on files, not directories '%s'", file))
+	end
+
+	---@type table<string, string>
+	local results = {}
+	local package = parser.get_first_match_string(file, package_query)
+	local classes = parser.get_all_matches_as_string(file, class_query)
+
+	for _, class in ipairs(classes) do
+		results[package .. "." .. class] = file
+	end
+
+	return results
+end
+
+---Determines all fully qualified classes in the provided path
+---@param path string
+---@return table<string, string>
+function M.determine_all_classes(path)
```

Comment:
This is probably the biggest change, now instead of using the path passed in (which only works for individual files) it's going over all files that exist and creating a table of

fully_qualified_class_name = path

This allows for better handling in `parse_test_id` too which is pretty nice